BaseClasses.py

```diff
@@ -1357,34 +1357,80 @@ def create_playthrough(self, create_paths: bool = True) -> None:
         # in the second phase, we cull each sphere such that the game is still beatable,
         # reducing each range of influence to the bare minimum required inside it
         restore_later: Dict[Location, Item] = {}
+        required_spheres: List[Tuple[Location, ...]] = []
         for num, sphere in reversed(tuple(enumerate(collection_spheres))):
-            to_delete: Set[Location] = set()
-            for location in sphere:
+            for location in tuple(sphere):
```

Comment:
It might be good to include a comment on why this `tuple` is here, so readers don't have to wonder about it for a minute like I did.